src/snapshot/component_clone.rs

```diff
@@ -0,0 +1,84 @@
+use crate::{
+    GgrsComponentSnapshot, GgrsSnapshots, LoadWorld, LoadWorldSet, Rollback, RollbackFrameCount,
+    SaveWorld, SaveWorldSet,
+};
+use bevy::prelude::*;
+use std::marker::PhantomData;
+
+/// A [`Plugin`] which manages snapshots for a [`Component`] `C` using [`Clone`].
+pub struct GgrsComponentSnapshotClonePlugin<C>
+where
+    C: Component + Clone,
+{
+    _phantom: PhantomData<C>,
+}
+
+type Snapshots<C> = GgrsSnapshots<C, GgrsComponentSnapshot<C>>;
+
+impl<C> Default for GgrsComponentSnapshotClonePlugin<C>
+where
+    C: Component + Clone,
+{
+    fn default() -> Self {
+        Self {
+            _phantom: Default::default(),
+        }
+    }
+}
+
+impl<C> GgrsComponentSnapshotClonePlugin<C>
+where
+    C: Component + Clone,
+{
+    pub fn save(
```

Comment:
Just a style thing, but I haven't often seen bevy systems as associated functions, is this a new thing. Is there an advantage or reason over just using a plain generic function?

Reply:
I mostly just did it for tidiness, since I can reuse the `where` clause across the whole type. I don't think it makes any difference under the hood.